app/models/manageiq/providers/amazon/refresh_helper_methods.rb

```diff
@@ -41,6 +41,17 @@ def add_instance_disk(disks, size, name, location)
     super(disks, size, name, location, "amazon")
   end
 
+  def add_block_device_disk(disks, name, location)
+    disk = {
+      :device_name     => name,
+      :device_type     => "disk",
+      :controller_type => "amazon",
+      :location        => location,
+    }
+    disks << disk
+    return disk
```

Comment:
@bronaghs I had a dillema :-). The code is similar to the one in the core repo: https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/refresh_parser.rb#L51. The reason for introducing a new function in the AWS provider is because we'd like not to mess something up in the core refresh parser for other managers (discussed with @Ladas).

One thing I will inspect, though, is whether OpenStack suffers from the same problem as AWS does. The code is very similar so I think it could, but I need a test.